internal/jwks/jwks.go

```diff
@@ -248,18 +251,41 @@ func (mgr *Manager) UpdateAll(ctx context.Context) error {
 	defer logger.Debug("finished jwks update")
 
 	haDB := hadb.New(mgr.db)
-	healthAuthorities, err := haDB.ListAllHealthAuthoritiesWithoutKeys(ctx)
+	allHealthAuthorities, err := haDB.ListAllHealthAuthoritiesWithoutKeys(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to query db: %w", err)
 	}
 
+	// Pre-filter out health authorities where discovery isn't enabled.
+	healthAuthorities := make([]*model.HealthAuthority, 0, len(allHealthAuthorities))
+	for _, ha := range allHealthAuthorities {
+		if ha.JWKS() {
+			healthAuthorities = append(healthAuthorities, ha)
+		}
+	}
+
+	// Crypto secure random isn't necessary here, just shuffling the scheduling order on each run.
+	rand.Shuffle(len(healthAuthorities), func(i, j int) {
+		healthAuthorities[i], healthAuthorities[j] = healthAuthorities[j], healthAuthorities[i]
+	})
+
 	var merr *multierror.Error
 	var merrLock sync.Mutex
-
+	sem := semaphore.NewWeighted(int64(mgr.maxWorkers))
```

Comment:
This can overflow. I'd suggest making the config an int and asserting its non-negative somewhere

Reply:
it's coming from a uint - the lint was because I was already converting it somewhere else